test/tests/functional/pbs_wean_udp.py

```diff
@@ -0,0 +1,124 @@
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        self.pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        self.exec_path = os.path.join(self.pbs_conf['PBS_EXEC'], "bin")
+
+    def submit_job(self, set_attrib=None, exp_attrib=None, sleep=10,
+                   interactive=False):
+        j = Job(PBSROOT_USER)
+        if set_attrib:
+            j.set_attributes(set_attrib)
+        if interactive:
+            j.interactive_script = [('hostname', '.*'),
+                                    ('export PATH=$PATH:%s' %
+                                     self.exec_path, '.*'),
+                                    ('qstat', '.*')]
+        j.set_sleep_time(sleep)
+        jid = self.server.submit(j)
+        if exp_attrib:
+            self.server.expect(JOB, exp_attrib, id=jid)
+        return jid
+
+    def submit_resv(self, set_attrib=None, exp_attrib=None):
+        r = Reservation(PBSROOT_USER)
+        if set_attrib:
+            r.set_attributes(set_attrib)
+        rid = self.server.submit(r)
+        if exp_attrib:
+            self.server.expect(RESV, exp_attrib, id=rid)
+        return rid
+
+    def test_comm_with_mom(self):
+        """
+        Test the installation of communication daemon during PBS installation.
+        """
+        msg = "Need 2 moms as input. use -pmoms=<m1>:<m2>"
+        if len(self.moms) != 2:
+            self.skip_test(reason=msg)
+        (self.momA, self.momB) = self.moms.values()
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+        log_msg = ["TPP initialization done",
+                   "Single pbs_comm configured, " +
+                   "TPP Fault tolerant mode disabled",
+                   "Connected to pbs_comm %s.*:17001" % self.server.shortname]
+        for msg in log_msg:
+            self.server.log_match(msg, regexp=True)
+            self.scheduler.log_match(msg, regexp=True)
+            self.momA.log_match(msg, regexp=True)
+            self.momB.log_match(msg, regexp=True)
+        server_ip = socket.gethostbyname(self.server.hostname)
+        msg = "Registering address %s:15001 to pbs_comm" % server_ip
+        self.server.log_match(msg)
+        msg = "Registering address %s:15004 to pbs_comm" % server_ip
+        self.scheduler.log_match(msg)
+        for mom_name in self.moms.values():
+            ip = socket.gethostbyname(mom_name.shortname)
+            msg1 = "Registering address %s:15003 to pbs_comm" % ip
+            msg2 = "Leaf registered address %s:15003" % ip
+            mom_name.log_match(msg1)
+            self.comm.log_match(msg2)
+        # Submit job
+        jid = self.submit_job(exp_attrib={'job_state': 'R'})
+        self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
+        self.server.log_match("%s;Exit_status=0" % jid)
+        # Submit multi-chunk job
+        set_attrib = {'Resource_List.select': '2:ncpus=1'}
```

Comment:
I suggest to add 'scatter' placement to this job in order to convert it to multi node job with pbsdsh.